src/care/algorithm_impl.h

```diff
@@ -979,6 +979,20 @@ CARE_INLINE void fill_n(care::host_device_ptr<T> arr, Size n, const U& val)
    } CARE_STREAM_LOOP_END
 }
 
+/************************************************************************
+ * Function  : copy_n
+ * Author(s) : Alan Dayton
+ * Purpose   : Copies one ManagedArray into another.
+ * ************************************************************************/
+template <class T, class Size, class U>
+CARE_INLINE void copy_n(care::host_device_ptr<T> in, Size n,
```

Comment:
Can the in array be made const?

Reply:
Or on the other hand I don't think std::copy_n is const , maybe that change isn't needed?

Reply:
Const would be nice, if not in the signature, at least before the lambda capture.

Reply:
Just made it const.

Reply:
Looks like I need both a const and non-const overload (I just made the non-const overload call the const overload).